app/src/main/java/org/oppia/app/customview/interaction/NumberWithUnitsInputInteractionView.kt

```diff
@@ -0,0 +1,245 @@
+package org.oppia.app.customview.interaction
+
+import android.content.Context
+import android.util.AttributeSet
+import android.widget.EditText
+import org.json.JSONObject
+import org.oppia.app.model.Fraction
+import org.oppia.app.model.InteractionObject
+import org.oppia.app.model.NumberUnit
+import org.oppia.app.model.NumberWithUnits
+import org.oppia.app.model.UserAnswer
+import org.oppia.app.parser.StringToFractionParser
+import org.oppia.app.player.state.answerhandling.InteractionAnswerHandler
+import org.oppia.domain.util.normalizeWhitespace
+import java.lang.Double.parseDouble
+import java.util.*
+import java.util.regex.Pattern
+
+/** The custom EditText class for number input with units interaction view. */
+class NumberWithUnitsInputInteractionView @JvmOverloads constructor(
```

Comment:
Similar to the other interaction views, please move all of the non view-specific logic to its own file. And, in this case, I suggest adding dedicated tests for that utility. The parsing logic here is much more complicated than the other interactions, and ought to have dedicated tests beyond the ones verifying number with units interaction correctness for this custom view.

Also, please add a link to the Oppia web version of this for reference in the new class.

Reply:
moved to StringToNumberWithUnitsParser file